### PR TITLE
WIP Upload test results as AppVeyor artifact

### PR DESCRIPTION
**Pull request by devlead**
_**Don't merge**_
This is just to test perf uploading test results as artifacts instead of automatically via XUnit reporter.

Will be rebased and tested post #1199 merged.

### CODE REVIEW

File: build.cake [hunk 1]
```diff
@@ -279,6 +279,10 @@ Task("Upload-AppVeyor-Artifacts")
 {
     AppVeyor.UploadArtifact(parameters.Paths.Files.ZipArtifactPathDesktop);
     AppVeyor.UploadArtifact(parameters.Paths.Files.ZipArtifactPathCoreClr);
+    foreach(var testResult in GetFiles(parameters.Paths.Directories.TestResults + "/*"))
```
Comment (gep13) [inline build.cake]: Yeah, that isn't going to work.  There are going to be multiple files in this folder, including the OpenCover result, which you don't really want to be uploading :-)

Comment (gep13) [inline build.cake]: @devlead perhaps limit this to files ending with `.Tests.xml`?

Comment (devlead) [inline build.cake]: yes will do.

**Comment by devlead** [comment]
AppVeyor working in fix, will reopen new PR to test once deployed,

**Pull request closed without merging**